### PR TITLE
.Net GoogleAI/VertexAI Gemini Text & Chat generation impl

### PR DESCRIPTION
**Pull request by Krzysztof318**
### Description

State of connector progress https://github.com/microsoft/semantic-kernel/issues/4680
Spit from PR https://github.com/microsoft/semantic-kernel/pull/4407

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiTextGenerationClient.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+using System.Net.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+
+/// <summary>
+/// Represents a client for interacting with the text generation gemini models by Vertex AI.
+/// </summary>
+// todo: remove this class when gemini vertex ai (preview api) support non-streaming chat completion
+internal sealed class VertexAIGeminiTextGenerationClient : GeminiTextGenerationClient, IGeminiTextGenerationClient
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiTextGenerationClient.cs:12]: This is a empty class for now, if there's more code to be added, that's fine, if not may consider removing it.

File: dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiEndpointProvider.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    /// </summary>
+    public Uri BaseEndpoint => new($"https://{this._configuration.Location}-aiplatform.googleapis.com/v1/projects/{this._configuration.ProjectId}/locations/{this._configuration.Location}/publishers/google/");
+
+    /// <summary>
+    /// Gets the endpoint URI for accessing the models in the Gemini API.
+    /// </summary>
+    public Uri ModelsEndpoint => new(this.BaseEndpoint, "models/");
+
+    /// <inheritdoc />
+    public Uri GetTextGenerationEndpoint(string modelId)
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiEndpointProvider.cs:34]: nit.

If the informations regarding the endpoints is always the same, we can move that information to the property and not creating new string instances everytime the method is invoked. Not a big deal, but something to be considered.

```suggestion
    public Uri GetTextGenerationEndpoint(string modelId)
```

File: dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiChatCompletionClient.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+
+/// <summary>
+/// Represents a client for interacting with the chat completion gemini models by Vertex AI.
+/// </summary>
+// todo: remove this class when gemini vertex ai (preview api) support non-streaming
+internal sealed class VertexAIGeminiChatCompletionClient : GeminiChatCompletionClient
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiChatCompletionClient.cs:17]: Are we going to have other specializations of GeminiChatCompletionClient ? 
Is VertexAIGemini completely different from other model supported by Google?

**Review by RogerBarreto — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiChatCompletionClient.cs (hunk 3 above)
Comment (Krzysztof318) [reply dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiChatCompletionClient.cs:17]: VertexAI should have same implementation like GoogleAI only other endpoints and requests. But current version of VertexAI rest api (preview) has limitations, so this is only reason why classes VertexAIGeminiTextGenerationClient and VertexAIGeminiChatCompletionClient exists. I intentionally added TODO comments to these classes. I hope before I finish connector, vertex ai will be stable and I will able to remove this classes completely.
I can delete VertexAIGeminiTextGenerationClient but VertexAIGeminiChatCompletionClient have to exist, at least for now.

File: dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiEndpointProvider.cs (hunk 2 above)
Comment (Krzysztof318) [reply dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiEndpointProvider.cs:34]: I just did some small optimalization in this class, but I prefer leave passing modelId and creating new Uri Instance as is now. It's more convenient than passing modelId and EmbeddingModelId to EndpointProvider contructor.

**Pull request merged**